packages/perseus/src/widgets/categorizer/categorizer.test.ts

```diff
@@ -145,4 +148,87 @@ describe("categorizer widget", () => {
         // Assert
         expect(userInput).toEqual({values: [1, 0, 0, 0, 1, 1]});
     });
+
+    describe("interactive: full vs answerless", () => {
+        beforeAll(() => {
+            registerAllWidgetsForTesting();
+        });
+
+        let userEvent: UserEvent;
+        beforeEach(() => {
+            userEvent = userEventLib.setup({
+                advanceTimers: jest.advanceTimersByTime,
+            });
+
+            jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+                testDependencies,
+            );
+        });
+
+        function getAnswerfulItem(): PerseusRenderer {
+            return {
+                content: "[[☃ categorizer 1]]",
+                images: {},
+                widgets: {
+                    "categorizer 1": {
+                        type: "categorizer",
+                        options: {
+                            static: false,
+                            items: ["Circle", "Purple", "Square"],
+                            categories: ["Shape", "Color"],
+                            randomizeItems: false,
+                            values: [0, 1, 0],
+                        },
+                    },
+                },
+            };
+        }
+
+        function getAnswerlessItem(): PerseusRenderer {
+            return splitPerseusItem(getAnswerfulItem());
+        }
+
+        test.each(["answerless", "answerful"])(
+            "is interactive with widget options: %p",
+            async (e) => {
+                // Arrange
+                const useAnswerless = e === "answerless";
+                const renderItem = useAnswerless
+                    ? getAnswerlessItem()
+                    : getAnswerfulItem();
```

Comment:
I'd prefer it if we could avoid conditionals in tests. If someone changes the `"answerless", "answerful"` strings here:

```ts
test.each(["answerless", "answerful"])( // ...
```

the tests will continue to pass, but they won't be testing what they're supposed to!

We could instead put the `item` in the test parameters:

```ts
test.each([
    ["answerless", getAnswerlessItem()],
    ["answerful", getAnswerfulItem()],
])(
    "is interactive with %s widget options",
    async (_, item) => {
        // ...
    }
)
```

Reply:
Sounds good to me! Updated. :)